lsp/nls/src/linearization/mod.rs

```diff
@@ -121,11 +121,13 @@ impl<'a> Linearizer for AnalysisHost<'a> {
                 .take()
                 .map(|(record, mut fields)| (record, fields.pop().unwrap()))
             {
-                pos = field_pos.map(|mut pos| {
-                    pos.start = ByteIndex(0);
-                    pos.end = ByteIndex(0);
-                    pos
-                });
+                if pos == TermPos::None {
+                    pos = field_pos.map(|mut x| {
+                        x.start = ByteIndex(u32::MAX);
+                        x.end = ByteIndex(u32::MAX);
+                        x
+                    });
+                }
```

Comment:
Why does this change fix the issue? It's not very clear to me. May I suggest to add something in the PR's description?

Side question, why do we need to fill a dummy position? I think we can now support terms without position. Maybe this code dates from before such support. How does it work if we just keep the `TermPos::None` value?

Reply:
First of all, we only want to add dummy position when the current position is a `TermPos::None`; the former code assigns this dummy position to every term's position (regardless of if it is a `TermPos::None` or not). This assigns the wrong position to nested record field's value. To avoid this, this was added on line 144: `pos = field.pos` (It assigns the position of a nested record field's value to be the same as the nested record field). However, we won't need to do this if we only give a dummy position to terms which have a position of `TermPos::None`.

Now, the problem with giving a record field (i.e the identifier) and it's value the same position is that when we want to find references at the position of a record field, we do not get the linearization item for the record field, but we get the linearization item for the record's field's value; so they should not be the same.

And finally, the reason why the dummy value was changed from `(0,0)` to `(max, max)` is that if we have another item at say, `(0, 9)`, and we want to get the item at `(0,9)` with `linearization.item_at((0,9))`, somehow, we get the item at `(0,0)`, so I think, `(max, max)` would be safer.

Reply:
>Side question, why do we need to fill a dummy position? I think we now support terms without position. Maybe this code dates back from before such support. How does it work if we just keep the TermPos::None value?

There's [this](https://github.com/tweag/nickel/blob/master/lsp/nls/src/linearization/mod.rs#L170), which doens't allow terms with `TermPos::None` to be linearized. Apart from that, [this](https://github.com/tweag/nickel/blob/master/lsp/nls/src/requests/symbols.rs#L30) expects terms to have a non-`TermPos::None` position.

Reply:
Hmm. Setting a position offset that is out of the file sounds wrong, I wonder if it could backfire (e.g. when doing error reporting or so). Maybe not, I guess the position inside the linearization is just used to find an item given a position. But I was convinced that the linearizer was able to handle stuff without position, by augmenting the ordering relation to `TermPos::None`. I'm surprised to see that in fact it can't.

Could we set the position of elaborated records correctly, as discussed during the daily? Could we also make the code work even if some position is set to `None`.